.github/actions/pr-review/pr_review.py

```
@@ -1532,8 +1585,222 @@ def _line_context(content: str, line: int | None) -> str:
+        rendered.append(piece)
+        used_tokens += estimate_tokens(piece)
+    if truncated:
+        rendered.append("<evidence-search-truncated: use unresolved unless the evidence above already decides the premise>")
+    return "\n".join(rendered), False
+
+
+def _bounded_git_grep(root: Path, term: str) -> tuple[list[str], bool, bool]:
+    """Read repository search output with hard time and byte bounds."""
+    try:
+        process = subprocess.Popen(  # noqa: S603
+            ["git", "-C", str(root), "grep", "-n", "-I", "-i", "-F", "-m", "3", "-e", term, "HEAD", "--"],
```

Ruff (0.16.1)

[error] Starting a process with a partial executable path

(S607)